csrc/host_ir/executor.h

```diff
@@ -13,30 +13,59 @@
 #include <host_ir/host_ir.h>
 #include <multidevice/communicator.h>
 #include <runtime/executor.h>
+#include <runtime/executor_abstract.h>
+#include <runtime/executor_params.h>
 #include <runtime/fusion_executor_cache.h>
 
 #include <c10/cuda/CUDAStream.h>
 
 namespace nvfuser {
 
+class HostIrExecutor : public ExecutorAbstract {
+ public:
+  HostIrExecutor(
+      int64_t fusion_id = 0,
+      int64_t concrete_id = 0,
+      int64_t runtime_id = 0,
+      int64_t group_id = 0);
+
+  static bool supported(Fusion* fusion);
+
+  void compile(Fusion* fusion);
+
+  bool isCompiled() const override;
+
+  NVF_API std::vector<at::Tensor> run(
+      KernelArgumentHolder& args,
+      std::vector<at::Tensor> outputs = {});
+
+  const std::unique_ptr<hir::HostIrContainer>& hostContainer() {
```

Comment:
Is this method needed? IIUC the host container shouldn't completely private.

Anyway, I would suggest
```suggestion
  const hir::HostIrContainer* hostContainer() {
```

@wujingyue for confirmation

Reply:
I don't recall writing this function, I wonder if it was just copy-pasted from `FusionExecutor` maybe. I'll need to check.

Reply:
I did write this function, and I think it was in the spirit of KernelExecutor being able to return its Fusion:
https://github.com/NVIDIA/Fuser/blob/main/csrc/runtime/executor.h#L195
or Kernel:
https://github.com/NVIDIA/Fuser/blob/main/csrc/runtime/executor.h#L190

I should make the function const. These functions are typically used in testing and I imagine would serve a similar purpose here.

Reply:
As long as it is `const` I guess it is ok